charts/nginx-gateway-fabric/templates/deployment.yaml

```diff
@@ -29,6 +29,33 @@ spec:
         {{- end }}
       {{- end }}
     spec:
+      initContainers:
+      - name: copy-nginx-config
+        image: {{ .Values.nginxGateway.image.repository }}:{{ default .Chart.AppVersion .Values.nginxGateway.image.tag }}
+        imagePullPolicy: {{ .Values.nginxGateway.image.pullPolicy }}
+        command:
+        - /usr/bin/gateway
+        - copy
+        - --source
+        - /includes/main.conf
+        - --destination
+        - /etc/nginx/main-includes/main.conf
```

Comment:
trying to understand the what this is doing. Since the name indicated `copy-nginx-config` shouldn't we copying `/etc/nginx/nginx.conf` ?

Reply:
This is only copying the config that we need at startup that we also need our control plane to be able to overwrite later if a user changes the log level. The base `nginx.conf` currently is not overwritten by our control plane, so we can just leave that alone and it can remain as read-only.

Does that make sense?

Reply:
The PR description explains a little more about the initial problem with files being read only and why we need to mount and then copy to a new location.

Reply:
Yes makes sense. Thank you. Was just a little confused with what exactly we were copying over due to the naming. I still recommend we name it related to base config, cause from my understanding of the above scenario it is a base config that will be overwritten by the control plane later during execution. Okay with what you decide.

Reply:
oh sorry for not being clear, meant that for the template name `copy-nginx-conf`

Reply:
Oh I see, I'll consider that on my next PR.